src/devices/bus/segaai/soundbox.cpp

```diff
@@ -0,0 +1,320 @@
+
+ICs on board:
+
+IC 2       TMP82C53F-2    (91/09)  Toshiba (Peripheral Timer)
+IC 3       HN27512G-25    (87/12)  Hitachi 64K EPROM
+IC 6       YM2151         (91/10)  Yamaha FM chip
+IC 7       TMP82C55AF-10  (88/15)  Toshiba (Peripheral Interface)
+IC 8       YM3012         (91/10)  Yamaha Stereo DAC
+IC 9       HA17358                 Hitachi Dual Op-Amp
+IC 10      LC7537N                 Sanyo (Volume Control IC)
+IC 11      C324C          (90/42)  NEC Quad Op-Amp
+IC 12      LA4520                  (Sanyo Power Audio Amp?)
+IC 16-19   MB81464-12     (91/12)  Fujitsu 32K DRAMs
+
+
+Misc Flat DIPs
+
+IC ??      LS125A        Hitachi (near C41)
+IC ??      74HC04        TI      (near C38)
+IC ??      74HC157A x2   Toshiba (near C37)
+IC ??      74HC138       TI      (near C44, furthest)
+IC ??      74HC139       TI      (near C44, closest)
+
+TODO:
+- Connections of the 8253
+- Keyboard matrix is scanned on a timer irq (#FC) from 8253??
+- LC7537N
+
+
+HC04
+pin 1 A1   - PB7
+pin 2 Y1   -> HC04 pin 3 A2
+pin 3 A2   <- HC04 pin 2 Y1
+pin 4 Y2   -> 4th point, 1st row below HC04?
+pin 5 A3   -
+pin 6 Y3   -
+pin 7 GND  -
+pin 8 Y4   -  1st point, 2nd row below HC04
+pin 9 A4   -  1st point, 1st row below HC04
+pin 10 Y5  -
+pin 11 A5  <- HC04 pin 12 Y6
+pin 12 Y6  -> HC04 pin 11 A5
+pin 13 A6  -  point just below C38 then continues to DRAMs
+pin 14 VCC -
+
+8255 PB7 - connected to HC04 pin 1?, pulled low
+
+TMP8253
+pin 9 CLK0 - seems to be tied to pin 24 in ym2151
+pin 14 OUT0 - --> 2nd point, 2nd row below HC04
+pin 15 GATE0 - NC
+pin 18 OUT1  - 7th point, 1st row below HC04 -> 8th point, 1st row below HC04 -> LS125 pin 2?
+pin 19 GATE1 - 6th point, 1st row below HC04 -> 4th point, 1st row below HC04
+pin 20 CLK1  - 5th point, 1st row below HC04 -> 2nd point, 1st row below HC04 -> left point above C37 -> pin 1 2 lc157s to the right of IC16 (can't be right)
+
+timer 0 - mode 3 - square wave (000A), gate not involved
+timer 1 - mode 2 - rate generator (0E90), gate involved
+0e90 = 3818
+
+ ***********************************************************************************************************/
+
+
+#include "emu.h"
+#include "soundbox.h"
+
+#include "machine/i8255.h"
+#include "machine/pit8253.h"
+#include "sound/ymopm.h"
+
+#include "speaker.h"
+
+
+namespace {
+
+class segaai_soundbox_device : public device_t,
+								public segaai_exp_interface
+{
+public:
+	segaai_soundbox_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock)
+		: device_t(mconfig, SEGAAI_SOUNDBOX, tag, owner, clock)
+		, segaai_exp_interface( mconfig, *this )
+		, m_tmp8253(*this, "tmp8253")
+		, m_tmp8255(*this, "tmp8255")
+		, m_ym2151(*this, "ym2151")
+		, m_rom(*this, "soundbox")
+		, m_rows(*this, "ROW%u", 0U)
+		, m_row(0)
+	{ }
+
+	virtual void device_add_mconfig(machine_config &config) override;
+	virtual const tiny_rom_entry *device_rom_region() const override;
+	virtual ioport_constructor device_input_ports() const override;
```

Comment:
Aren’t these overridden protected member functions?  They aren’t part of the public interface.

Reply:
Made these member functions protected again.